interview-questions.md

```diff
@@ -8,48 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 3. What is a React component?
 
-Your answer:
+Your answer: A React component is one piece of an entire React application. React components allow for specific, defined functions to reside in. Each React component is independent of one another, but essentially are invoked in a main component. That main component is typically called App.js where in best practice is where all logic for an app resides and component invocation. Data can be passed to components using props. Data can also be passed from a component to the main component using prop functions. React components are typically designed to be reusable for all kinds of data. 
 
-Researched answer:
+Researched answer: React components are mostly made up of JS code and importantly they return markup language to the browser. Components mostly are comprised of functions that become reusable over and over again. 
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: React state values are always being updated. When a useState is initially setup, it will have a default static value to start. But, as functionality is implemented in components, the useState will be updated by the defined setter method. React props are data being passed into components. This data is usually the same type and is not updated constantly like states values. 
 
-Researched answer:
+Researched answer: State values go hand and hand with what components do logically. If a component is holding data then it would be considered a logic component. If it does not hold data, it would be considered a display component. useState is a React hook which allows for built-in methods for components to be reusable in the future. It is very important to remember to import your useState so that it can be initialized and updated properly. State values can be referenced in JSX to be rendered. Props(properties) allows data to be passed between components. Props get passed within component invocation or within the component function. 
 
 5. What is the DOM?
 
-Your answer:
+Your answer: Document Object Model is the framework for which React is structured. 
 
-Researched answer:
+Researched answer: DOM is essentially what the user is looking at in the browser. When working with React, whichever component you are working on, only that piece will be rendered on the browser side of the application. 
 
 6. STRETCH: Which is the difference between a div tag and a span tag?
 
-Your answer:
+Your answer: An HTML div tag is a general container tag. Container tags can contain class names that can be referenced in CSS files. span tags are another HTML container tags but, allow different functionality. 
 
-Researched answer:
+Researched answer: Both div and span tags are general HTML container tags. div tags are used for larger content and take up space on the entire browser page. span tags are used for in-line content and allow for CSS styling to certain parts of content.  
```

Comment:
You could also try to provide an example. Div tags will also allow for CSS styling to certain parts of content. By providing an example, you become more specific with your answer and really get to show what you know. 

This is what a span tag allows you to do!
If we wrapped the words "span tag" in a span tag and gave it style color blue and font weight bold... 
Then, span tag would be bold and blue in that sentence.